### DISCUSSION TITLE
TypeSpec in the same repository or a different one?

### DISCUSSION THREAD

**Discussion by sfre-coop** [microsoft/typespec#3400] [Q&A] [answered]
I'm experimenting with TypeSpec for a service we're providing, implemented in node. Should I include the TypeScript definitions in the implementation repository, including it in the build and keeping everything in sync, or should I use a separate repository and generate a node library to be referenced?

Thanks all.

**Comment by bterlson** [comment, marked as answer]
I think TypeSpec doesn't particularly care whether you put the spec in another repo, same repo but different package in a monorepo setup, or in the same package. I would probably avoid a separate repo just because it will introduce a fair amount of friction when iterating on the spec and iteration together, but either of the other two would be fine.

**Reply by sfre-coop** [reply]
Thanks. Has anyone written up adding to an existing package?
